gammapy/makers/background/reflected.py

```diff
@@ -41,6 +155,35 @@ def run(self, region, center, exclusion_mask=None):
         wcs: `~astropy.wcs.WCS`
             WCS for the determined regions
         """
+        reference_geom = self._create_reference_geometry(region, center)
+        center_pixel = self._get_center_pixel(center, reference_geom)
+
+        region_pix = self._get_region_pixels(region, reference_geom)
+        excluded_pixels = self._get_excluded_pixels(reference_geom, exclusion_mask)
+
+        n_positions = self.n_off_regions + 1
+        increment = FULL_CIRCLE / n_positions
+
+        regions = []
+        for i in range(1, n_positions):
+            angle = i * increment
+            region_test = region_pix.rotate(center_pixel, angle)
+
+            if exclusion_mask is None or not np.any(region_test.contains(excluded_pixels)):
+                regions.append(region_test)
+
+
+        # check for overlaps
```

Comment:
This could use https://github.com/gammapy/gammapy/blob/master/gammapy/maps/wcs/ndmap.py#L547 to simplify the code...

Reply:
mmh. I don't think so. That is for testing if a sky region is included in a sky mask. We are still in pixel space here and the code is pretty straight forward, right? Converting every region repeatedly into the list of pixel coordinates seems also more expensive than calculating the masks once and then checking logical and

Reply:
Sorry, my bad. This indeed checks for the overlap of the regions themselves and not for the overlap with the exclusion mask.

Reply:
In this case maybe move the check inside the other for loop? I think it's sufficient to check whether the `region_test`, overlaps with any of the previous regions and then break?

Reply:
One could also extend the excluded pixels by the previously found regions, but that's maybe a bit too clever...

Reply:
Yes I thought about that, but like that you will end up with a single region that might overlap with the on region. Actually we should probably just be able to check if the first region overlaps with the on region, right?

Reply:
> Actually we should probably just be able to check if the first region overlaps with the on region, right?

Hm, if the offset is really large and the number of requested off regions high, one could still end up with overlapping off regions, that don't overlap with the on region?

Reply:
I dont think so, since the basis of the code is that the off regions and the on region are symmetrically distributed. If I don't tell you the on region, you should not be able to tell.

Reply:
Yes, what I said before doesn't make sense. I had the wrong geometry of the problem in mind...

The symmetry argument is key here. In this case I agree it should be sufficient to check whether the first off region overlaps with the on region and that's it.